code/modules/defenses/defenses.dm

```diff
@@ -144,6 +144,8 @@
 			faction_group = FACTION_LIST_HUMANOID
 		if(SENTRY_FACTION_COLONY)
 			faction_group = list(FACTION_MARINE, FACTION_COLONIST)
+		if(FACTION_UPP)
```

Comment:
I would suggest also adding an option in here, however you see fit, to set IFF exclusively to WY & PMCs, without including Marines.

Aside from that, all code looks good to me. Sprites are great, though I can't review those.